project/auth/routes.py

```diff
@@ -0,0 +1,179 @@
+# Modules from Flask Library
+from flask import Flask, render_template, request, flash, Blueprint
+from flask import session as login_session
+from flask import make_response, redirect, jsonify, url_for
+# Access database
+from project.models import Base, Category, Product, User
+from project.db import DBConnector
+# Services
+from project.services.categories import CategoryService
+from project.services.config import ConfigService
+
+import random
+import string
+import httplib2
+import json
+import requests
+
+# Authentification modules
+from oauth2client.client import flow_from_clientsecrets
+from oauth2client.client import FlowExchangeError
+
+
+auth = Blueprint('auth', __name__)
+
+# Connect to the database, create session
+session = DBConnector().get_session()
+
+# Instantiate services
+category_service = CategoryService()
+config_service = ConfigService()
+
+# Get Client ID
+CLIENT_ID = config_service.get_setting('client_id')
+
+
+@auth.route('/login')
+def showLogin():
+    """App route function to display login page."""
+    state = ''.join(random.choice(string.ascii_uppercase + string.digits)
+                    for x in range(32))
+    login_session['state'] = state
+    return render_template('login.html',
+                           STATE=state,
+                           categories=category_service.get_all_categories(),
+                           CLIENT_ID=CLIENT_ID)
+
+
+@auth.route('/googleConnect', methods=['POST'])
+def googleConnect():
+    """Connect via Google Account and fetch User info."""
+    # Validate state token
+    if request.args.get('state') != login_session['state']:
+        response = make_response(json.dumps('Invalid state parameter.'), 401)
+        response.headers['Content-Type'] = 'application/json'
+        return response
+    # Obtain authorization code
+    code = request.data
+
+    try:
+        # Upgrade the authorization code into a credentials object
+        oauth_flow = flow_from_clientsecrets('client_secrets.json', scope='')
+        oauth_flow.redirect_uri = 'postmessage'
+        credentials = oauth_flow.step2_exchange(code)
+    except FlowExchangeError:
+        response = make_response(
+            json.dumps('Failed to upgrade the authorization code.'), 401)
+        response.headers['Content-Type'] = 'application/json'
+        return response
+
+    # Check that the access token is valid.
+    access_token = credentials.access_token
+    url = ('https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=%s'
+           % access_token)
```

Comment:
Nit: nowadays, it's a good habit to use `.format()` instead of `%` for string formatting. For ex:
```suggestion
url = 'https://www.googleapis.com/oauth2/v1/tokeninfo?access_token={}'.format(
            access_token,
        )```